content/tutorials/parallelization/SIMWE_parallelization.qmd

```diff
@@ -0,0 +1,804 @@
+```
+
+Run hydrologic and sediment simulations for the selected subwatershed.
+First, simulate overland flow using [r.sim.water](https://grass.osgeo.org/grass-devel/manuals/r.sim.water.html)
+with inputs for topography, Manning’s coefficients, and rainfall intensity.
+Running the simulation may take a while.
+
+```{python}
+simulation_time = 180
+tools.r_sim_water(
+    elevation="ned",
+    dx=f"dx_{huc12}",
+    dy=f"dy_{huc12}",
+    depth=f"depth_{huc12}",
+    niterations=simulation_time,
+    man=f"mannings_{huc12}",
+    rain=f"runoff_{huc12}",
+)
+
+basin_map = gj.Map()
+basin_map.d_rast(map=f"depth_{huc12}")
+basin_map.d_legend(
+    raster=f"depth_{huc12}",
+    flags="t",
+    at=[10, 15, 40, 95],
+    title="Depth [m]",
+    fontsize=12,
+)
+basin_map.show()
+```
+
+![](SIMWE_images/basin_030401010402_simwe.webp)
+
+Then, define parameters for sediment transport and run [r.sim.sediment](https://grass.osgeo.org/grass-devel/manuals/r.sim.sediment.html)
+to compute erosion and deposition patterns. We will trim the resulting raster's edges to avoid extreme values at the edge.
+The additional parameters for sediment erosion modeling are based on [WEPP](https://www.ars.usda.gov/midwest-area/west-lafayette-in/national-soil-erosion-research/docs/wepp/research/) model, here we use just a single-value estimate.
+
+```{python}
+tools.r_mapcalc(expression=f"transport_capacity_{huc12} = 0.001")
+tools.r_mapcalc(expression=f"detachment_coef_{huc12} = 0.001")
+tools.r_mapcalc(expression=f"shear_stress_{huc12} = 0.01")
+tools.r_sim_sediment(
+    elevation="ned",
+    water_depth=f"depth_{huc12}",
+    dx=f"dx_{huc12}",
+    dy=f"dy_{huc12}",
+    detachment_coeff=f"detachment_coef_{huc12}",
+    transport_coeff=f"transport_capacity_{huc12}",
+    shear_stress=f"shear_stress_{huc12}",
+    niterations=simulation_time,
+    erosion_deposition=f"erdep_tmp_{huc12}",
+    nwalkers=5 * gs.region()["cells"],
+)
+tools.r_grow(input=f"erdep_tmp_{huc12}", output=f"erdep_{huc12}", radius=-2.01)
+
+basin_map = gj.Map()
+basin_map.d_rast(map=f"erdep_{huc12}")
+basin_map.d_legend(
+    raster=f"erdep_{huc12}",
+    flags="t",
+    at=[10, 15, 40, 95],
+    range=[-0.5, 0.5],
+    digits=2,
+    title="Erosion/dep. [kg/m2s]",
+    fontsize=12,
+)
+basin_map.show()
+```
+
+![](SIMWE_images/basin_030401010402_erdep.webp)
+
+Calculate total erosion for the subwatershed.
+Negative values from [r.sim.sediment](https://grass.osgeo.org/grass-devel/manuals/r.sim.sediment.html)
+output are treated as erosion and converted to sediment mass.
+Summary statistics are computed using [r.univar](https://grass.osgeo.org/grass-devel/manuals/r.univar.html).
+
+```{python}
+# Erosion: extract negative values and convert to positive mass [kg/m^2]
+tools.r_mapcalc(
+    f"erosion_{huc12} = if(erdep_{huc12} < 0, abs(erdep_{huc12}) * {simulation_time * 60}, null())"
+)
+erosion = tools.r_univar(map=f"erosion_{huc12}", format="json")
+```
+
+```text
+ 'null_cells': 418835,
```

Comment:
`...` or `{` missing.

```suggestion
```text
...
 'null_cells': 418835,
```

It might be also little confusing to show the a result like this which is actually pretty printed Python dict, but the code does not suggests printing and suggests JSON (which has slightly different syntax).